client/components/tinymce/plugins/wpcom-view/wpvideo-view.jsx

```diff
@@ -0,0 +1,137 @@
+
+/**
+ * Internal dependencies
+ */
+import shortcodeUtils from 'lib/shortcode';
+import { getVideo } from 'state/videos/selectors';
+import QueryVideo from 'components/data/query-video';
+
+class WpVideoView extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	constrainVideoDimensions( shortcodeWidthAttribute, shortcodeHeightAttribute ) {
+		let { width, height } = this.props.video
+			? this.props.video
+			: {
+				width: 640,
+				height: 640 / ( 16 / 9 )
+			};
+		const aspectRatio = width / height;
+
+		if ( shortcodeWidthAttribute && ! shortcodeHeightAttribute ) {
+			width = shortcodeWidthAttribute;
+			height = shortcodeWidthAttribute / aspectRatio;
+		} else if ( ! shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			width = shortcodeHeightAttribute * aspectRatio;
+			height = shortcodeHeightAttribute;
+		} else if ( shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			width = shortcodeWidthAttribute;
+			height = shortcodeHeightAttribute;
+		}
+
+		return { width, height };
+	}
+
+	getVideoAttributes() {
+		const shortcode = shortcodeUtils.parse( this.props.content );
+		const namedAttrs = shortcode.attrs.named;
+
+		const videoDimensions = this.constrainVideoDimensions(
+			parseInt( namedAttrs.w, 10 ),
+			parseInt( namedAttrs.h, 10 )
+		);
+
+		const shortcodeAttributes = {
+			guid: shortcode.attrs.numeric[0],
+			w: videoDimensions.width,
+			h: videoDimensions.height,
+			autoplay: includes( [ 'true', '1' ], namedAttrs.autoplay ),
+			hd: includes( [ 'true', '1' ], namedAttrs.hd ),
+			loop: includes( [ 'true', '1' ], namedAttrs.loop ),
+			at: parseInt( namedAttrs.at, 10 ) || 0,
+			defaultLangCode: namedAttrs.defaultlangcode
+		};
+
+		return Object.assign(
+			pick( shortcodeAttributes, [ 'guid', 'w', 'h' ] ),
+			{ embedUrl: this.getEmbedUrl( shortcodeAttributes ) }
+		);
+	}
+
+	getEmbedUrl( shortcodeAttributes ) {
+		const defaultAttributeValues = { hd: false, at: 0, defaultLangCode: undefined };
+		const attributesWithNonDefaultValues = omitBy(
+			shortcodeAttributes,
+			( value, key ) => defaultAttributeValues[ key ] === value );
+		const queryStringAttributes = [ 'autoplay', 'hd', 'loop', 'at', 'defaultLangCode' ];
+		const queryString = QueryString.stringify( pick( attributesWithNonDefaultValues, queryStringAttributes ) );
+
+		return `https://videopress.com/embed/${ shortcodeAttributes.guid }?${ queryString }`;
+	}
+
+	onLoad() {
+		const doc = this.props.editor.iframeElement.contentDocument;
+		const script = doc.createElement( 'script' );
+		script.src = 'https://videopress.com/videopress-iframe.js';
+		script.type = 'text/javascript';
+		doc.getElementsByTagName( 'head' )[0].appendChild( script );
```

Comment:
Each event handler is executed but only one does real work, the others return early.

`detect` is the `DOMContentLoaded` event handler, which is smart enough not to attach resize logic to the same VideoPress iframe twice. Relevant part of the code is the `el__vp` property setting/reading:

```js
function detect() {
        var frames = document.getElementsByTagName('iframe');
        for (var i = 0; i < frames.length; i++) {
            var el = frames[i];
            if (el.__vp) continue;
            if (isVideoPress(el)) {
                attach(el);
            }
        }
    }

    function attach(el) {
        var parent = el.parentNode;
        if (!parent) return;
        var initialWidth = el.width;
        var initialHeight = el.height;
        var resize = function resize() {
            el.width = Math.min(parent.offsetWidth, initialWidth);
            el.height = initialHeight * el.width / initialWidth;
        };
        resize();
        __vpResize(el, resize);
        el.__vp = true;
    }
```